finality-aleph/src/sync/service.rs

```diff
@@ -374,10 +357,20 @@ where
 
     /// Stay synchronized.
     pub async fn run(mut self) {
+        use NetworkData::*;
         loop {
             tokio::select! {
                 maybe_data = self.network.next() => match maybe_data {
-                    Ok((data, peer)) => self.handle_network_data(data, peer),
+                    Ok((data, peer)) => match data {
+                        StateBroadcast(state) => self.handle_state(state, peer),
```

Comment:
Imo it's better to have this match as a method rather than so much nesting.

Reply:
fixed